Android/app/src/main/java/jgrrw/painttheworld/LoginActivity.java

```diff
@@ -0,0 +1,277 @@
+                switchToMaps.putExtra(GRIDSIZE_LONGITUDE, gridsizeLongitude);
+                switchToMaps.putExtra(GRIDSIZE_LATITUDE, gridsizeLatitude);
+                startActivity(switchToMaps);
+                pollingHandler.removeCallbacks(pollToSeeIfGameHasStarted);
+            }
+            else {
+                pollingHandler.postDelayed(pollToSeeIfGameHasStarted, 500);
+            }
+        }
+    };
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_login);
+
+
+        //Get location
+        if (ContextCompat.checkSelfPermission(this,
+                android.Manifest.permission.ACCESS_FINE_LOCATION)
+                != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(this,
+                    new String[]{android.Manifest.permission.ACCESS_FINE_LOCATION},
+                    MY_PERMISSIONS_REQUEST_ACCESS_FINE_LOCATION);
+        }
+        getLocation();
+
+        b1 = (Button) findViewById(R.id.JoinButton);
+        b = findViewById(R.id.JoinButton);
+        tx1 = (TextView) findViewById(R.id.AboutText);
+        tx1.setTextSize(30);
+        tx1.setText("Loading...");
+    }
+
+    //Methods for Google Play Services Location
+    private void getLocation() {
+        locationRequest = LocationRequest.create();
+        locationRequest.setPriority(LocationRequest.PRIORITY_HIGH_ACCURACY);
+        locationRequest.setInterval(LOCATION_INTERVAL);
+        locationRequest.setFastestInterval(LOCATION_INTERVAL);
+        fusedLocationProviderApi = LocationServices.FusedLocationApi;
+        mGoogleApiClient = new GoogleApiClient.Builder(this)
+                .addApi(LocationServices.API)
+                .addConnectionCallbacks(this)
+                .addOnConnectionFailedListener(this)
+                .build();
+        if (mGoogleApiClient != null) {
+            mGoogleApiClient.connect();
+        }
+    }
+
+    @Override
+    public void onLocationChanged(android.location.Location location) {
+        locationIsValid = true;
+        if (!buttonVisible) {
+            b.setVisibility(View.VISIBLE);
+            buttonVisible = true;
+        }
+        tx1.setText("\nHello!\n\nWelcome to\nPaintTheWorld!");
+        mLocation.set(location);
+        Log.d("LOCATION:", mLocation.getLatitude() + ", " + mLocation.getLongitude());
+    }
+
+
+    public void joinGame(View view) throws JSONException
+    {
+        if (!locationIsValid) {
+            return;
+        }
+        final String URL = "http://ec2-54-153-39-233.us-west-1.compute.amazonaws.com/join_lobby";
+        // Post params to be sent to the server
+        HashMap<String, Double> params = new HashMap<String, Double>();
+        params.put("lat", mLocation.getLatitude());
+        params.put("long", (mLocation.getLongitude()));
+
+        JsonObjectRequest req = new JsonObjectRequest(URL, new JSONObject(params),
```

Comment:
True, but I like hashmap for readability purposes.  I'll consider changing this in the future.